src/application.ts

```diff
@@ -96,14 +99,14 @@ export default class Application {
     const processor = processors.find(p => p !== false);
 
     if (processor) {
-      return new processor(this);
+      return new processor(new ApplicationInstance(this));
```

Comment:
I don't think this is actually doing things correctly?

Seems like you're getting a new application instance per processor not per bulk...?

Reply:
Yeah, this might not be a good idea.